keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md

```diff
@@ -0,0 +1,1079 @@
+This change reduces the attack surface and the risk of credential leaks, aligning with
+best practices for security.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+To keep this KEP small both in terms of implementation and scope,
+the design is written to meet the following criteria:
+
+- No changes to any Kubernetes REST APIs
+- No changes to the Kubernetes API server
+- No changes to the CRI API
+- No changes to how the Kubelet interacts with the CRI API
+- No changes to any registry
+- Minimal changes to Kubelet
+- Minimal changes to Kubelet credential providers
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+TODO(aramase)
+1. add details about the ensure image pulls KEP and impact on this KEP - Talk about how the caching logic will require changes.
+
+#### Credential Provider Configuration
+
+Add `Audience` field to `CredentialProvider` struct in `CredentialProviderConfig`.
+This field will be optional and if not set, no KSA token will be sent to the plugin.
+
+- Only a single audience can be configured.
+- The credential provider can be set up multiple times for each audience.
+
+```diff
+--- a/pkg/kubelet/apis/config/types.go
++++ b/pkg/kubelet/apis/config/types.go
+@@ -648,6 +648,11 @@ type CredentialProvider struct {
+        // to pass argument to the plugin.
+        // +optional
+        Env []ExecEnvVar
++
++       // Audience is the intended audience for the credentials. If set, the kubelet will
++       // generate a service account token for the audience and pass it to the plugin.
++       // +optional
++       Audience string
+```
+
+For the pod to be able to use the projected service account token for image pulls, the pod spec must include a projected service account token volume with the audience set to the credential provider audience. 
+Only the volume is required in the pod spec, and it does not need to be mounted. The node audience restriction will be enforced by the [NodeRestriction admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#noderestriction)
```

Comment:
If the only reason for this is to inform kube-apiserver of the audiences the kubelet should be allowed to request, giving it that information directly (via config/invocation or dedicated API) seems better than a dummy unmounted volume.

Reply:
Added a new flag for this in KAS